### ISSUE TITLE
icinga2.state file corrupt after satellite restart

### ISSUE THREAD

**Issue by m4k5ym** [Icinga/icinga2#8645]
## Describe the bug

Following scenario is currently a huge butt pain.

```
        M1
    /         \
( SAT1  <>  SAT2 )
   |/        \|
( SAT3 )  ( SAT4 )
```

I have to update my SAT1 and SAT2 on operating system level. This mostly result in the fact that I have to reboot the machine because of kernel updates. So far so good.

But after SAT1 is done with reboot my whole Icinga 2 environment is bricked.

What does bricked exactly mean? Yes sir. No check is triggred anywhere in the chain. **EVER**.

The only solution I have found is to stop Icinga 2 daemon on M1, SAT1 and SAT2. Then I delete icinga2.state file and start M1 first, then SAT1 and SAT2 in the end. (if SAT1 or SAT2 first imo doesn't matter at all)

I mentioned **_butt pain_**. So what is happening if I delete the state file and restart Icinga 2 daemon? Correct! Master of scheduling is executing like 300 Host and 6200 Service checks at once what results in a huge explosion of nearly every Icinga 2 satellite and agent.

I can't count that as a side effect since a few satellites are requesting nearly 32GB memory and 20GHz for a restart just because of this whole "behaviour" and non existing scheduling.

## To Reproduce

I can't provide anything at the moment since I even don't understand what happens at all.

## Expected behavior

Not breaking.

## Screenshots

There is nothing to show in pictures.

## Your Environment

Include as many relevant details about the environment you experienced the problem in

* Version used (`icinga2 --version`): r2.12.3-1
* Operating System and version: Debian major 10 minor 8
* Enabled features (`icinga2 feature list`): api checker debuglog graphite ido-mysql mainlog
* Icinga Web 2 version and modules (System - About): not involved
* Config validation (`icinga2 daemon -C`):
```
[2021-02-22 10:46:24 +0100] information/cli: Icinga application loader (version: r2.12.3-1)
[2021-02-22 10:46:24 +0100] information/cli: Loading configuration file(s).
[2021-02-22 10:46:24 +0100] warning/config: Ignoring directory '/etc/icinga2/zones.d/11111-satellite-1' for unknown zone '11111-satellite-1'. (**shorted due to privacy reasons**)
[2021-02-22 10:46:24 +0100] warning/config: Ignoring directory '/etc/icinga2/zones.d/22222-satellite-1-off' for unknown zone '22222-satellite-1-off'. (**shorted due to privacy reasons**)
[2021-02-22 10:46:24 +0100] warning/config: Ignoring directory '/var/lib/icinga2/api/zones/11111-master' for unknown zone '11111-master'. (**shorted due to privacy reasons**)
[2021-02-22 10:46:24 +0100] warning/config: Ignoring directory '/var/lib/icinga2/api/zones/33333-dereg' for unknown zone '33333-dereg'. (**shorted due to privacy reasons**)
[2021-02-22 10:46:24 +0100] warning/config: Ignoring directory '/var/lib/icinga2/api/zones/22222-master' for unknown zone '22222-master'. (**shorted due to privacy reasons**)
[2021-02-22 10:46:24 +0100] warning/config: Ignoring directory '/var/lib/icinga2/api/zones/22222-satellite-1' for unknown zone '22222-satellite-1'. (**shorted due to privacy reasons**)
[2021-02-22 10:46:24 +0100] information/ConfigItem: Committing config item(s).
[2021-02-22 10:46:24 +0100] information/ApiListener: My API identity: master.localhost.localdomain (shorted due to privacy reasons)
[2021-02-22 10:46:26 +0100] warning/ApplyRule: Apply rule 'array-health' (in /etc/icinga2/zones.d/global-templates/service/global/global-centreon.conf: 59:1-59:28) for type 'Service' does not match anywhere!
[2021-02-22 10:46:26 +0100] warning/ApplyRule: Apply rule 'disk ' (in /etc/icinga2/zones.d/global-templates/service/global/global-disk-windows.conf: 5:1-5:60) for type 'Service' does not match anywhere!
[2021-02-22 10:46:26 +0100] warning/ApplyRule: Apply rule 'imap ' (in /etc/icinga2/zones.d/global-templates/service/global/global-imap.conf: 5:1-5:60) for type 'Service' does not match anywhere!
[2021-02-22 10:46:26 +0100] warning/ApplyRule: Apply rule 'load' (in /etc/icinga2/zones.d/global-templates/service/global/global-load-windows.conf: 5:1-5:20) for type 'Service' does not match anywhere!
[2021-02-22 10:46:26 +0100] warning/ApplyRule: Apply rule 'memory' (in /etc/icinga2/zones.d/global-templates/service/global/global-memory-windows.conf: 5:1-5:22) for type 'Service' does not match anywhere!
[2021-02-22 10:46:26 +0100] warning/ApplyRule: Apply rule 'int health ' (in /etc/icinga2/zones.d/global-templates/service/global/global-nwc_health.conf: 29:1-29:79) for type 'Service' does not match anywhere!
[2021-02-22 10:46:26 +0100] warning/ApplyRule: Apply rule 'procs ' (in /etc/icinga2/zones.d/global-templates/service/global/global-procs-windows.conf: 5:1-5:62) for type 'Service' does not match anywhere!
[2021-02-22 10:46:26 +0100] warning/ApplyRule: Apply rule 'update' (in /etc/icinga2/zones.d/global-templates/service/global/global-updates-windows.conf: 5:1-5:22) for type 'Service' does not match anywhere!
[2021-02-22 10:46:26 +0100] warning/ApplyRule: Apply rule 'uptime' (in /etc/icinga2/zones.d/global-templates/service/global/global-uptime.conf: 5:1-5:22) for type 'Service' does not match anywhere!
[2021-02-22 10:46:26 +0100] warning/ApplyRule: Apply rule 'vmware vc ' (in /etc/icinga2/zones.d/global-templates/service/global/global-vmware.conf: 17:1-17:92) for type 'Service' does not match anywhere!
[2021-02-22 10:46:26 +0100] information/ConfigItem: Instantiated 306 Hosts.
[2021-02-22 10:46:26 +0100] information/ConfigItem: Instantiated 1 GraphiteWriter.
[2021-02-22 10:46:26 +0100] information/ConfigItem: Instantiated 2 NotificationCommands.
[2021-02-22 10:46:26 +0100] information/ConfigItem: Instantiated 2 FileLoggers.
[2021-02-22 10:46:26 +0100] information/ConfigItem: Instantiated 12 Comments.
[2021-02-22 10:46:26 +0100] information/ConfigItem: Instantiated 224 Notifications.
[2021-02-22 10:46:26 +0100] information/ConfigItem: Instantiated 1 IcingaApplication.
[2021-02-22 10:46:26 +0100] information/ConfigItem: Instantiated 1 CheckerComponent.
[2021-02-22 10:46:26 +0100] information/ConfigItem: Instantiated 48 Zones.
[2021-02-22 10:46:26 +0100] information/ConfigItem: Instantiated 48 Endpoints.
[2021-02-22 10:46:26 +0100] information/ConfigItem: Instantiated 1 IdoMysqlConnection.
[2021-02-22 10:46:26 +0100] information/ConfigItem: Instantiated 2 ApiUsers.
[2021-02-22 10:46:26 +0100] information/ConfigItem: Instantiated 1 ApiListener.
[2021-02-22 10:46:26 +0100] information/ConfigItem: Instantiated 262 CheckCommands.
[2021-02-22 10:46:26 +0100] information/ConfigItem: Instantiated 2 TimePeriods.
[2021-02-22 10:46:26 +0100] information/ConfigItem: Instantiated 6 UserGroups.
[2021-02-22 10:46:26 +0100] information/ConfigItem: Instantiated 12 Users.
[2021-02-22 10:46:26 +0100] information/ConfigItem: Instantiated 6204 Services.
[2021-02-22 10:46:26 +0100] information/ScriptGlobal: Dumping variables to file '/var/cache/icinga2/icinga2.vars'
[2021-02-22 10:46:26 +0100] information/cli: Finished validating the configuration file(s).
```
* If you run multiple Icinga 2 instances, the `zones.conf` file (or `icinga2 object list --type Endpoint` and `icinga2 object list --type Zone`) from all affected nodes. 
```
#
# master
#

# primary, us
object Endpoint "master.localhost.localdomain" {
}

object Zone "master" {
  endpoints = [ "master.localhost.localdomain" ]
}

#
# satellite
#

# primary
object Endpoint "satellite1.localhost.localdomain" {
  host = "satellite1.localhost.localdomain"
  port = "5665"
  log_duration = 14d
}

# secondary
object Endpoint "satellite2.localhost.localdomain" {
  host = "satellite2.localhost.localdomain"
  port = "5665"
  log_duration = 14d
}

object Zone "satellite" {
  endpoints = [ "satellite1.localhost.localdomain", "satellite2.localhost.localdomain" ]
  parent = "master"
}

#
# global
#

object Zone "global-templates" {
  global = true
}
```

**Comment by julianbrost** [comment]
Can you please provide logs from the involved nodes? Also how did you figure out that the state file must be corrupt? Is there something in the logs?

**Comment by m4k5ym** [comment]
Hello,

This is actually not a great position of mine for debugging but unfortunately logs are exposing a huge amount of sensitive data. But its not first time that behavior occurs and I didn't find anything in logs which indicates something useful.

How I concluded that my state file is corrupted? As I wrote already if I restart first satellite everything stops working until I remove state file from master.

Also I have the feel that clustering of mine satellite isn't working properly. Normally SAT2 must overtake everything if SAT1 dies but I don't really believe this since most checks becoming overdue if SAT 1 dies.
